sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs

```diff
@@ -42,30 +42,44 @@ public async override Task<AuthenticationEventResponse> Completed()
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    // Response.MarkAsFailed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage));
+                    return await Failed(new AuthenticationEventTriggerRequestValidationException(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage)).ConfigureAwait(false);
```

Comment:
In what scenario would status message be empty? I don't like the idea of us throwing a generic "failed".

Reply:
I'm not sure what scenarios would give a null or empty status message. But I will do some investigation into this.

Reply:
This is a property that the end user can alter and possibly make it null or empty